Dockerfile.tests

```diff
@@ -7,19 +7,19 @@ ARG BUILDER_BASE=quay.io/cilium/cilium-envoy-builder:bazel-5.1.1@sha256:9e94c4c6
 ARG PROXYLIB=quay.io/cilium/cilium:stable
 
 #
-# BUILDER_IMAGE defaults to the result of the first stage below,
+# ARCHIVE_IMAGE defaults to the result of the first stage below,
 # refreshing the build caches from Envoy dependencies before the final
 # build stage. This can be overridden on docker build command line to
 # use pre-built dependencies. Note that if cross-compiling, these
 # pre-built dependencies will include BUILDPLATFORM build tools and
 # TARGETPLATFORM build artifacts, and thus can only be reused when
 # building on the same BUILDPLATFORM.
 #
-ARG BUILDER_IMAGE=builder-fresh
+ARG ARCHIVE_IMAGE=archive-cache
 
 FROM $PROXYLIB as proxylib
 
-FROM --platform=$BUILDPLATFORM $BUILDER_BASE as builder-fresh
+FROM --platform=$BUILDPLATFORM $BUILDER_BASE as archive-cache
```

Comment:
IMO this stage name should keep on the connotation that it is building from scratch (hence the "fresh").